nsls-ii-xlsheet/xlsheet_data.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+from bs4 import BeautifulSoup
+from time import sleep
+import requests
+import warnings
+import httplib2
+warnings.filterwarnings('ignore', category = UserWarning) # supresses the MarkupResemblesLocatorWarning(UserWarning) message for bs4
+
+def get_versions(namedata, URLv, filename):
+
+    for i in range(len(namedata)):
+        link = URLv + namedata[i].strip()
+        html = BeautifulSoup(link, features = 'lxml')
+        p = html.findAll('p')
+        type(p)
+        len(p)
+        p0=p[0]
+        r = requests.get(p0.text, timeout=100)
+        svg = BeautifulSoup(r.text, features = 'lxml')
+        version_tag = svg.findAll('text')[-1]
+
+        with open(filename, 'a') as file:
```

Comment:
You can get the information in a list or dict, and then just write it in one shot. Alternatively, the file can be opened once outside of the `for` loop.

Reply:
made a YAML file to keep the feedstock_URL, anaconda_URL, conda version, nsls2 version. 

kept the names.txt file since we use it to update the number and name of packages easily, and by using its file length i makes it easy for using loops.